examples/ansible/code/Synchronize_DNS_Zones

```diff
@@ -0,0 +1,50 @@
+- name: Sync DNS
+  hosts: <your Bind-Server>
+  tasks:
+    - name: get Managed Zones Names
+      ansible.builtin.set_fact:
+        zone_names: "{{ query('netbox.netbox.nb_lookup', 'zones', plugin='netbox_dns',
+                           api_endpoint='https://<nextcloud-url>/',
+                           api_filter='status=active status=deprecated',
+                           token='<redacted>')
+                    | map(attribute='value') | map(attribute='name') }}"
+
+    - name: Generate Zone Files
+      template:
+          src: dns.zone.j2
+          dest: "/path-to-your-zone-folder/{{ item }}.zone"
+      vars:
+          zonename: "{{ item }}"
+          zone: "{{ query('netbox.netbox.nb_lookup', 'zones', plugin='netbox_dns',
+                           api_endpoint='https://<nextcloud-url>/',
```

Comment:
Suggestion: Put things like the URL of the server and file paths into variables, then use them throughout the playbook.

Reply:
there i wasnt concentrated, it should just say netbox, not nextcloud

Reply:
When I suggested using variables I was thinking about something like this:
```
vars:
    netbox_api_endpoint: https://netbox.example.com
    netbox_api_token: xxxxxxxxxxxxxxxxxxxxxxxx
    zone_file_path: /tmp/bind

[...]
  - name: Generate Zone Files
      template:
          src: dns.zone.j2
          dest: "{{ zone_file_path }}/{{ item }}.zone"
      vars:
          zonename: "{{ item }}"
          zone: "{{ query('netbox.netbox.nb_lookup', 'zones', plugin='netbox_dns',
                           api_endpoint=netbox_api_endpoint,
                           api_filter='zone='+zone.name,
                           token=netbox_api_token)
                       | map(attribute='value') }}"
```
That has the advantage of having only one central point to adjust settings to individual values instead of spreading changes throughout the playbook.

Reply:
Yes that is already in my repository and tested